src/fields.jl

```diff
@@ -176,3 +179,13 @@ for ft in (:CellField, :FaceFieldX, :FaceFieldY, :FaceFieldZ, :EdgeField)
         end
     end
 end
+
+xnodes(c::CellField) = reshape(c.grid.xC, c.grid.Nx, 1, 1)
+ynodes(c::CellField) = reshape(c.grid.yC, 1, c.grid.Ny, 1)
+znodes(c::CellField) = reshape(c.grid.zC, 1, 1, c.grid.Nz)
+
+set!(u, a::AbstractArray) = u.data .= a
```

Comment:
These will have to be modified for `OffsetArrays` by using explicit loops. See the corresponding code for `OceanTurb`:

https://github.com/glwagner/OceanTurb.jl/blob/efc9e95a55ef4018231d9fb8282b956ccfdd2f2f/src/fields.jl#L148

Reply:
Yes. It was modified in PR #167.

https://github.com/climate-machine/Oceananigans.jl/blob/da7b26263d007dffee1d0ae94e18c4d97cd8f919/src/fields.jl#L160

Reply:
Nice. I guess broadcast + views is better than a loop because it works on CPU and GPU without needing GPUifyLoop magic.

Reply:
Yeah that was my thinking. `@views` allocates a bit of memory but we shouldn't be setting fields during time stepping anyway so the performance hit in favor of readability here should be okay.

Reply:
Heh. Actually for optimization / data assimilation problems we may actually need to do performance optimization on model initialization, field setting, etc. (for column modeling we are currently initializing 100,000s of models)

But we can dispatch on the array type of `Field` (once the array is added as type parameter to `Field`, which it certainly should) and achieve higher performance / less allocation if we need to.

Reply:
Ah fair enough, was not thinking of that use case.

I guess when focusing on the 1D model case there will be quite a few zeroth-order optimization to be made (e.g. not calculating _x_ and _y_-derivatives for 1D vertical models) but then things like this will start to matter and we can write optimized kernels for them.

Reply:
Interesting point. Also it’s a waste to use OffsetArray in the horizontal and make all fields 3x as big. I guess it’s good we currently have separate codes for running column models.

Reply:
Yup. The sizes of the halo regions are variable but part of what I need to polish off in PR #167 is making sure this kind of stuff happens for 2D and 1D models, which is where unit testing will help.